python-package/lightgbm/basic.py

```diff
@@ -2202,6 +2202,45 @@ def free_network(self):
     def trees_to_dataframe(self):
         """Parse the fitted model and return in an easy-to-read pandas DataFrame.
 
+        The returned DataFrame has the following columns.
+
+            - ``tree_index``: int64, which tree a node belongs to.
+              0-based, so a value of ``6``, for example, means "this node is in the
+              7th tree".
+            - ``node_depth``: int64, how far a node is from the root of the tree.
+              The root node has a value of ``1``, its direct children are ``2``, etc.
+            - ``node_index``: string, unique identifier for a node
+            - ``left_child``: string, ``node_index`` of the child node to the left
+              of a split. ``None`` for leaf nodes.
+            - ``right_child``: string, ``node_index`` of the child node to the right
+              of a split. ``None`` for leaf nodes.
+            - ``parent_index``: string, ``node_index`` of this node's parent. ``None``
+              for the root node.
+            - ``split_feature``: string, identifier for the feature used for splitting.
+              This is of the form ``"Column_i"``, where ``i`` refers to the number of the
+              feature. For example, the first feature would be ``"Column_0"``. ``None``
+              for leaf nodes.
```

Comment:
This description is not always true. Feature names can have custom values passed by user.
```suggestion
            - ``split_feature``: string, name of the feature used for splitting.
               ``None`` for leaf nodes.
```

Reply:
ok, I'll change this then. But I don't think just saying "name of the feature used for splitting" is sufficient. We should document both cases.

What do you think about this?

> string, name of the feature used for splitting. If `feature_name` was not provided to the `fit()` call that produced this model, this will be of the form `"Column_i"`, where `i` refers to the number of the feature. So for example, the first feature would be `"Column_0"`.

I think it's important to specify that these `Column_i` names are 0-based (since that might not be obvious), and that they're based on the order of the input data.

Reply:
@jameslamb 

> was not provided to the `fit()` call that produced this model

There are a lot of methods where feature names can be set, not only `fit()` from sklearn wrapper.

I believe it is enough to say that default feature names are in a form of `"Column_i"` (zero-based). But please my next paragraph.

>  We should document both cases.

Yeah, I support this idea! But I think that this can be documented in a more "general" place, because default feature names are common among all wrappers.

Reply:
oh great points! Ok I'll change it, I agree with you